### DISCUSSION TITLE
Generating configuration/documentation from BaseSettings

### DISCUSSION THREAD

**Discussion by radeklat** [pydantic/pydantic#5154] [Projects using Pydantic]
Hello everyone. I'd like to share a project I wrote some time ago with you all. It's a command line tool for generating configuration/documentation from one or more `BaseSettings` classes. At the moment, it can generate:

- Markdown documentation
- .env file

Since it is powered by the Jinja2 templating engine, it is easily extendable to other use cases, such as Kubernetes manifests. It also supports customizable heading/spacing offsets and in-place updates. So it can fit into existing files.

You can find it at https://github.com/radeklat/settings-doc.

As a side note, is there a way to link to it from the official documentation? If people find it useful, they will still have hard time finding it organically.

**Comment by merura** [comment]
10Q for your work! I love this tiny tool
